samza-core/src/main/java/org/apache/samza/execution/JobGraph.java

```diff
@@ -105,14 +109,11 @@ public String getPlanAsJson() throws Exception {
    * Returns the config for this application
    * @return {@link ApplicationConfig}
    */
+  @Override
   public ApplicationConfig getApplicationConfig() {
```

Comment:
Where is this method used? If this class doesn't mutate the config, would be nice to not expose the config from here.

Reply:
This is used in JobGraphJsonGenerator and RemoteJobPlanner. Also, defined as the public interface method in ExecutionPlan. Besides, the ApplicationConfig object is immutable as it extends Config.

Reply:
Not sure why this is part of the API though. Since JobNode already exposes the generated config, callers could probably do `new ApplicationConfig(generatedConfig)` and use it. If so, would prefer to remove this, since it's confusing to have many different getters for config that don't specify how they're different from each other.